compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -1717,7 +1717,7 @@ impl NodeInterner {
         // Replace each generic with a fresh type variable
         let substitutions = impl_generics
             .into_iter()
-            .map(|typevar| (typevar.id(), (typevar, self.next_type_variable())))
+            .map(|typevar| (typevar.id(), (typevar, Kind::Normal, self.next_type_variable())))
```

Comment:
I'm moving `TypeVariableKind` to `TypeBinding::Unbound` since
- When we have a `TypeBinding::Bound`, we can easily derive the `Kind`
- This ensure we'll always have the kind, i.e. allows `TypeVariable::kind(&self) -> Kind`